progress.go

```diff
@@ -92,9 +95,17 @@ func (p *Progress) Report(s Stat) {
 	p.curM.Lock()
 	p.cur.Add(s)
 	cur := p.cur
+	needUpdate := false
+	if time.Since(p.lastUpdate) > minTickerTime {
```

Comment:
how about extracting this into a new method?
```
func (p *Progress) needUpdate() { ... }
```
I think that would improve readability. And then use `if p.needUpdate() { ... }` below.